### ISSUE TITLE
coverage planner crashes PC

### ISSUE THREAD

**Issue by hs0747** [ethz-asl/polygon_coverage_planning#96]
if i try to create a polygon by calling the '/coverage_planner/set_polygon' service, coverage_planner stops at "Start creating sweep plan graph. After that RAM and SWAP memory fully loaded and crushes the PC. 

Also received this error.
Stack overflow in unmanaged: IP: 0x7f8e0c49ffff, fault addr: 0x7ffdd6beeff8
Stack overflow: IP: 0x7f8e0c49ff3a, fault addr: 0x7ffdd6be6ff8
Stacktrace:

![Screenshot from 2023-12-20 16-05-25](https://github.com/ethz-asl/polygon_coverage_planning/assets/113107715/30168c1d-59ca-45de-8c72-37a779000ddc)
![Screenshot from 2023-12-20 16-05-21](https://github.com/ethz-asl/polygon_coverage_planning/assets/113107715/aedceab3-7285-4da7-8413-491f3c904a5f)
![Screenshot from 2023-12-20 16-05-11](https://github.com/ethz-asl/polygon_coverage_planning/assets/113107715/39d7e8c4-46aa-4128-949d-9fdc5f97641c)

**Comment by Ivanfangsc** [comment]
Hello, did you find a fix for this problem?

**Comment by ircbloqcc** [comment]
Nope, i did a python script to monitor the polygon coverage. if the ram memory of polygon coverage exceeds more than 2GB, i used to kill the rosnode.

That the way i managed this problem

**Comment by hs0747** [comment]
Nope, i did a python script to monitor the polygon coverage. if the ram memory of polygon coverage exceeds more than 2GB, i used to kill the rosnode.

That the way i managed this problem
